python/xorq/cli.py

```diff
@@ -114,8 +120,72 @@ def run_command(expr_path, output_path=None, output_format="parquet"):
             raise ValueError(f"Unknown output_format: {output_format}")
 
 
+@tracer.start_as_current_span("cli.serve_command")
+def serve_command(
+    expr_path, host=None, port=None, duckdb_path=None, prometheus_port=None
+):
+    """
+    Serve a built expression via Flight Server
+
+    Parameters
+    ----------
+    expr_path : str
+        Path to the expression directory (output of xorq build)
+    host : str
+        Host to bind Flight Server
+    port : int or None
+        Port to bind Flight Server (None for random)
+    duckdb_path : str or None
+        Path to duckdb cache DB file
+    """
+
+    span = trace.get_current_span()
+    params = {
+        "build_path": expr_path,
+        "host": host,
+        "port": port,
+    }
+    if duckdb_path is not None:
+        params["duckdb_path"] = duckdb_path
+    span.add_event("serve.params", params)
+
+    expr_path = Path(expr_path)
+    expr_hash = expr_path.stem
+
+    logger.info(f"Loading expression '{expr_hash}' from {expr_path}")
+    build_manager = BuildManager(expr_path.parent)
+    if not build_manager.artifact_store.exists(expr_hash, "expr.yaml"):
+        raise ValueError(f"Error: expr.yaml not found in build directory {expr_path}")
+
+    expr = build_manager.load_expr(expr_hash)
+
+    db_path = Path(duckdb_path or "xorq_serve.db")  # what should be the default?
+
+    db_path.parent.mkdir(parents=True, exist_ok=True)
+    logger.info(f"Using duckdb at {db_path}")
+
+    try:
+        from xorq.flight.metrics import setup_console_metrics
+
+        setup_console_metrics(duckdb_path=str(db_path), prometheus_port=prometheus_port)
+    except ImportError:
+        logger.warning(
+            "Metrics support requires 'opentelemetry-sdk' and console exporter"
+        )
+
+    def _make_con():
+        return xo.duckdb.connect(str(db_path))
+
+    server = FlightServer.from_udxf(expr, connection=_make_con)
```

Comment:
rather than defining `_make_con`, we can
```
from functools import partial

server = FlightServer.from_udxf(expr, connection=partial(xo.duckdb.connect, str(db_path)))
```

Reply:
related: lets standardize the name to `make_connection` or similar in `FlightServer`, `FlightServerDelegate` and whatever else gets passed this value